ClientApp/viking-vault/src/RegisterForm.tsx

```diff
@@ -0,0 +1,233 @@
+        this.setState({
+            email: e.target.value
+        }, this.validateEmail);
+    }
+
+    private emailError = () => {
+        if(!emailRegEx.test(this.state.email)) {
+            return  {
+                msg: "Email format incorrect",
+                class: "alert alert-danger"
+            }
+        }
+        if(this.state.valid === "invalid") {
+            return  {
+                msg: "Email already exists",
+                class: "alert alert-danger"
+            }
+        }
+        if(this.state.valid === "valid") {
+            return  {
+                msg: "Valid",
+                class: "alert alert-success"
+            }
+        }
+        return  {
+            msg: "Server error. Try again later",
+            class: "alert alert-danger"
+        }
+    }
+
+    private handleCPasswordChange = (e: any) => {
+        this.setState({
+            cpassword: e.target.value
+        }, this.passwordMatch)
+    }
+
+    private passwordMatch = () => {
+        if(this.state.password === this.state.cpassword)
+            return  {
+                msg: "Passwords match",
+                class: "alert alert-success"
+            }
+        return  {
+            msg: "Password do not match",
+            class: "alert alert-danger"
+        }
+    }
+
+    private mandatoryFieldsCompletedCorrectly = () => {
+        let val = true;
+        if(this.emailError().msg !== "Valid") {
+            val = false;
+        }
+        if(this.passwordMatch().msg !== "Passwords match") {
+            val = false;
+        }
+        if(this.state.address === "" || this.state.cnp === "" || this.state.firstName === "" 
+            || this.state.lastName === "" || this.state.password === "") {
+                val = false;
+        }
+        return val;
+    }
+
+    private execute = async () => {
+        this.setState({
+            openModal: true,
+            response: "Please wait..."
+        });
+        const user = {
+            Email: this.state.email,
+            Password: this.state.password,
+            FirstName: this.state.firstName,
+            LastName: this.state.lastName,
+            Address: this.state.address,
+            Cnp: this.state.cnp,
+            PictureLink: this.state.pictureLink
+        };
+        fetch(url+"user", {
+            method: "POST",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(user)
+        },).then(response => {
+            if(response.status !== 200) {
+                this.setState({
+                    response: "Error. Please Try again",
+                    openModal: true
+                });
+            }
+            return response.json();
+        }).then(result => {
+            setTimeout(() => {
+                this.setState({
+                    redirect: true
+                });
+            }, 1500);
```

Comment:
Are you adding an artificial delay to the redirect?

Reply:
Yes :D

Reply:
Why?